src/components/timeline/Ruler.js

```diff
@@ -19,45 +20,66 @@ type Props = {|
 |};
 
 export class TimelineRuler extends PureComponent<Props> {
-  _findNiceNumberGreaterOrEqualTo(uglyNumber: number) {
+  _findNiceNumberGreaterOrEqualTo(uglyNumber: Milliseconds) {
+    // Special case numbers in the seconds, minutes or hour ranges.
+    if (uglyNumber > 10000 && uglyNumber <= 48 * 3600 * 1000) {
+      for (const seconds of [15, 20, 30]) {
+        const number = seconds * 1000;
+        if (uglyNumber <= number) {
+          return number;
+        }
+      }
+      for (const minutes of [1, 2, 5, 10, 15, 20, 30]) {
+        const number = minutes * 60 * 1000;
+        if (uglyNumber <= number) {
+          return number;
+        }
+      }
+      for (const hours of [1, 2, 3, 4, 6, 8, 12, 24, 48]) {
+        const number = hours * 3600 * 1000;
+        if (uglyNumber <= number) {
+          return number;
+        }
+      }
```

Comment:
Maybe I'm missing something, but it looks like we could just return number here instead of doing all these loops? Indeed you're not returning anything specific from the loops and the various conditions...

Reply:
The point of these loops is to decide which value to assign in the 'number' variable. The possible values for 'number' are pre-defined by the values in the arrays. Then the first 'number' value that's < ugglyNumber is what we keep (ie. return).